src/TourManager/TourManager.Repository.EfCore.MsSql/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using TourManager.Repository.Abstraction;
+
+namespace TourManager.Repository.EfCore.MsSql.Repositories
+{
+    public class BaseRepository<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        protected readonly DbContext Context;
+        protected readonly DbSet<TEntity> DbSet;
+
+        public BaseRepository(DbContext context)
+        {
+            Context = context;
+            DbSet = Context.Set<TEntity>();
+        }
+
+        public async Task<TEntity> GetByIdAsync(int id)
+        {
+            return await Context.Set<TEntity>().FindAsync(id);
+        }
+
+        public async Task<IEnumerable<TEntity>> GetAllAsync()
+        {
+            var query = Context.Set<TEntity>().AsNoTracking();
+
+            return await query.ToListAsync();
+        }
+
+        public async Task<IEnumerable<TEntity>> Find(Expression<Func<TEntity, bool>> predicate)
```

Comment:
There is no need to create async method just return task and await in place of usage

Reply:
yeah, you're right

Reply:
done